modules/git/foreachref/parser.go

```diff
@@ -30,6 +30,11 @@ type Parser struct {
 func NewParser(r io.Reader, format Format) *Parser {
 	scanner := bufio.NewScanner(r)
 
+	// default MaxScanTokenSize = 64 kiB may be too small for some references,
+	// so allow the buffer to grow up to 4x if needed
+	buf := make([]byte, 0, bufio.MaxScanTokenSize)
+	scanner.Buffer(buf, 4 * bufio.MaxScanTokenSize)
```

Comment:
For most cases, we do not need a that large buffer, neither `bufio.Scanner` does so.

Scanner uses 4096 as the initial size:

> startBufSize = 4096 // Size of initial allocation for buffer.


```suggestion
	// default MaxScanTokenSize = 64 kiB may be too small for some references,
	// so allow the buffer to grow up to 4x if needed
	scanner.Buffer(nil, 4 * bufio.MaxScanTokenSize)
```